### DISCUSSION TITLE
gao.opentelemetry.io/otel/metric@upgrade found (v0.36.0), but does not contain package go.opentelemetry.io/otel/metric/instrument/asyncfloat64

### DISCUSSION THREAD

**Discussion by lzap** [open-telemetry/opentelemetry-go#3740] [Q&A] [answered]
Hello,

I am updating deps of our project, but getting:

```
go: all: module go.opentelemetry.io/otel/metric@upgrade found (v0.36.0), but does not contain package go.opentelemetry.io/otel/metric/instrument/asyncfloat64
go: all: module go.opentelemetry.io/otel/metric@upgrade found (v0.36.0), but does not contain package go.opentelemetry.io/otel/metric/instrument/asyncint64
go: all: module go.opentelemetry.io/otel/metric@upgrade found (v0.36.0), but does not contain package go.opentelemetry.io/otel/metric/instrument/syncfloat64
go: all: module go.opentelemetry.io/otel/metric@upgrade found (v0.36.0), but does not contain package go.opentelemetry.io/otel/metric/instrument/syncint64
```

Any idea how to solve this?

https://github.com/RHEnVision/provisioning-backend/blob/main/go.mod

Is my understanding correct that one of our dependency still depends on syncint64?

```
% go mod why go.opentelemetry.io/otel/metric/instrument/syncint64
# go.opentelemetry.io/otel/metric/instrument/syncint64
github.com/RHEnVision/provisioning-backend/internal/clients/http
go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp
go.opentelemetry.io/otel/metric/instrument/syncint64
```

We apparently use otelhttp package which needs syncint64. Thanks for help.

**Comment by Petrie** [comment, marked as answer]
Package `go.opentelemetry.io/otel/metric/instrument/{asyncfloat64,asyncint64,syncfloat64,syncint64}` is deprecated at [v0.35.0](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.12.0), and remove at [v0.36.0](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.13.0).

You need replace them with the new API.  See details [v0.35.0](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.12.0). 
You can specify an older version number in you `go.mod` if  don't want to upgrade your code immediately.

**Reply by lzap** [reply]
I am trying to understand what is causing this, we do not import `otel/metric` directly and what I do is full module update, so I would expect that `contrib` package is also updated to match the latest version. Thanks @Petrie for help!

**Reply by lzap** [reply]
Hmm I am on Go 1.18 at the moment, that is what our build environment use and I can't do anything about it.

Specifying the older version works for me, thank you.

```
lzap@mone provisioning-backend % go get -u go.opentelemetry.io/otel/metric/instrument@v0.35.0
go: downloading go.opentelemetry.io/otel/metric v0.35.0
go: downloading go.opentelemetry.io/otel v1.12.0
go: upgraded go.opentelemetry.io/otel v1.11.2 => v1.13.0
go: upgraded go.opentelemetry.io/otel/metric v0.34.0 => v0.35.0
go: upgraded go.opentelemetry.io/otel/trace v1.11.2 => v1.13.0
```

**Reply by lzap** [reply]
The problem was I was doing `go get -u all` and for some reason it did not work. Looks like `go get -u ./...` works just fine.
